pyspeckit/spectrum/fitters.py

```diff
@@ -412,14 +412,25 @@ def EQW(self, plot=False, plotcolor='g', fitted=True, continuum=None,
         if plot and self.Spectrum.plotter.axis:
             if midpt_location == 'plot-center':
                 midpt_pixel = np.round((xmin+xmax)/2.0)
-                midpt       = self.Spectrum.xarr[midpt_pixel]
+                midpt       = self.Spectrum.xarr[midpt_pixel].value
             elif midpt_location == 'fitted':
                 try:
-                    midpt = sp.specfit.parinfo.SHIFT0.value
+                    PI_keys = self.Spectrum.specfit.parinfo.keys()
+                    shifts = [self.Spectrum.specfit.parinfo[PI_keys[x]].value for x in range(1, len(PI_keys), 3)]
```

Comment:
It would be better to have a check like:
`shifts = [self.Spectrum.specfit.parinfo[x].value for x in self.Spectrum.specfit.parinfo.keys() if 'SHIFT' in x]`, i.e. check for the word `SHIFT` rather than relying on the numerical order of the parameters.